ta/pkcs11/src/pkcs11_helpers.c

```diff
@@ -479,6 +479,42 @@ bool pkcs2tee_load_attr(TEE_Attribute *tee_ref, uint32_t tee_id,
 	return true;
 }
 
+/*
+ * Initialize a TEE attribute with hash of a target PKCS11 TA attribute
+ * in an object.
+ */
+bool pkcs2tee_load_hashed_attr(TEE_Attribute *tee_ref, uint32_t tee_id,
+			       struct pkcs11_object *obj,
+			       enum pkcs11_attr_id pkcs11_id, uint32_t tee_algo,
+			       void *hash_ptr, uint32_t *hash_size)
+{
+	TEE_OperationHandle handle = TEE_HANDLE_NULL;
+	void *a_ptr = NULL;
+	uint32_t a_size = 0;
+	TEE_Result res = TEE_ERROR_GENERIC;
+
+	if (get_attribute_ptr(obj->attributes, pkcs11_id, &a_ptr, &a_size))
+		return false;
+
+	res = TEE_AllocateOperation(&handle, tee_algo, TEE_MODE_DIGEST, 0);
+	if (res) {
+		EMSG("TEE_AllocateOperation() failed %#"PRIx32, tee_algo);
+		return false;
+	}
+
+	res = TEE_DigestDoFinal(handle, a_ptr, a_size, hash_ptr, hash_size);
+	TEE_FreeOperation(handle);
+	if (res) {
+		EMSG(" TEE_DigestDoFinal() failed %#"PRIx32, tee_algo);
+		return false;
+	}
+
+	TEE_InitRefAttribute(tee_ref, tee_id, hash_ptr, *hash_size);
+
+	return true;
+}
+
+
```

Comment:
remove empty line

Reply:
Sorry, I didn't notice it earlier. Have removed it.